vendor/models/profile.py

```diff
@@ -109,4 +112,13 @@ def get_cart_items_count(self):
 
     def get_or_create_address(self, address):
         address, created = self.addresses.get_or_create(name=address.address_1, first_name=address.first_name, last_name=address.last_name, address_1=address.address_1, address_2=address.address_2, locality=address.locality, state=address.state, country=address.country, postal_code=address.postal_code, profile=self)
-        return address, created
\ No newline at end of file
+        return address, created
+
+    def has_previously_owned_products(self, products):
+        """
+        Get all products that the customer has purchased and returns True if it has.
+        """
+        return bool(self.receipts.filter(Q(products__in=products), Q(status__gte=PurchaseStatus.COMPLETE)).first())
```

Comment:
Question for anyone: why use `Q()` here? does the filter not already AND for you?

@mgcamisa @rhimmelbauer @renderbox

Reply:
No need to add the Q, probably was using it for another query and forgot to update it properly thanks @devo-wm